src/snowflake/cli/_plugins/nativeapp/commands.py

```diff
@@ -475,10 +482,24 @@ def app_events(
     ),
     **options,
 ):
-    """Fetches events for this app from the event table configured in Snowflake."""
+    """Fetches events for this app from the event table configured in Snowflake.
+
+    By default, this command will fetch events generated by an app installed in the
+    current connection's account. To fetch events generated by an app installed
+    in a consumer account, use the --consumer-org and --consumer-account flags.
+    This requires event sharing to be set up to route events to the provider account:
+    https://docs.snowflake.com/en/developer-guide/native-apps/setting-up-logging-and-events
+    """
     if first >= 0 and last >= 0:
         raise ClickException("--first and --last cannot be used together.")
 
+    if (consumer_org and not consumer_account) or (
+        consumer_account and not consumer_org
+    ):
+        raise ClickException(
```

Comment:
would UsageError() be more accurate in this case?

Reply:
good idea, updated all the checks in this command